srml/system/src/lib.rs

```diff
@@ -76,16 +76,20 @@ use serde::Serialize;
 use rstd::prelude::*;
 #[cfg(any(feature = "std", test))]
 use rstd::map;
-use primitives::weights::{Weight, WeightMultiplier};
-use primitives::{generic, traits::{self, CheckEqual, SimpleArithmetic,
+use rstd::marker::PhantomData;
+use primitives::generic::{self, Era};
+use primitives::weights::{Weight, DispatchInfo, DispatchClass, WeightMultiplier};
+use primitives::transaction_validity::{ValidTransaction, TransactionPriority, TransactionLongevity};
+use primitives::traits::{self, CheckEqual, SimpleArithmetic, Zero, SignedExtension, Convert,
 	SimpleBitOps, Hash, Member, MaybeDisplay, EnsureOrigin, CurrentHeight, BlockNumberToHash,
-	MaybeSerializeDebugButNotDeserialize, MaybeSerializeDebug, StaticLookup, One, Bounded, Lookup,
-	Zero, Convert,
-}};
+	MaybeSerializeDebugButNotDeserialize, MaybeSerializeDebug, StaticLookup, One, Bounded,
+	Lookup, DispatchError, SaturatedConversion,
+};
+use primitives::Perbill;
```

Comment:
This currently unused because: I want to get rid of all the `max_weight / 4` and make it a `Perbill` associated const in system.

Reply:
`type IdealWeightRatio = Get<Perbill>`

and wrap the multiplication in a public function.

Reply:
Will be done in another PR. first, want to check if it is the correct way to go since it requires a daunting refactor.